### PR TITLE
Add IdentityX newsletter components

### PR DESCRIPTION
**Pull request by B77Mills**
**REQUIRES: https://github.com/parameter1/identity-x/pull/16**
 - Add ability to set successMessageType(login || newsletter-signup)
 - Add ability to send forceProfileReVerification in additionalEventData which will force the user to submit their profile on next login
Screen Shot 2022-06-21 at 2 02 23 PM
Screen Shot 2022-06-21 at 2 02 38 PM

### CODE REVIEW

**Comment by solocommand** [comment]
Updated implementation allows for reusability by moving components into Monorail theme, with backwards compatibility for RR (new components, existing ones stay as-is). To implement, use the new `identity-x-newsletter-form-inline`, `identity-x-newsletter-form-footer` (along with new `@newsletter-block` attribute for theme footer) and `identity-x-newsletter-form-pushdown` (in place of "theme-site-newsletter-menu").

|Initial|Submitted|Logged in |
|-|-|-|
| screencapture-www-p1-sandbox-dev-parameter1-51269-directory-news-22288695-lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-2022-06-27-12_52_28 | screencapture-www-p1-sandbox-dev-parameter1-51269-directory-news-22288695-lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-2022-06-27-12_53_27 | screencapture-www-p1-sandbox-dev-parameter1-51269-directory-news-22288695-lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-2022-06-27-12_54_12 |

**Review by solocommand — DISMISSED** [review]

**Comment by B77Mills** [comment]
@solocommand  my last two commits I added a idx-newsletter-toggle component that works with the idx-pushdown component.  This allows for everything to be independent from the current newsletter stuff.  I also adjusted the css for the inline component.

**Screen shots of them loaded side by side:**
 - **Pushdown** Screen Shot 2022-06-28 at 9 51 11 AM
 - **Content Inline** Screen Shot 2022-06-28 at 9 50 59 AM
 - **Section Inline** Screen Shot 2022-06-28 at 9 51 26 AM

File: packages/marko-web-theme-monorail/components/site-header.marko [hunk 1]
```diff
@@ -62,3 +63,12 @@ $ const navigation = {
...
 </marko-web-block>
+
+<marko-web-block>
+  <if(useIdxNewsletterSignup)>
+      <identity-x-newsletter-form-pushdown />
+    </if>
+    <else>
+      <theme-site-newsletter-menu />
+    </else>
+</marko-web-block>
```
Comment (solocommand) [inline packages/marko-web-theme-monorail/components/site-header.marko]: @B77Mills Indentation is off here. Is there a reason you're wrapping this in a marko-web-block tag? That doesn't seem to be done within the site-specific implementations...would just remove it unless it's necessary for some reason.

**Review by solocommand — CHANGES_REQUESTED** [review]

File: packages/marko-web-theme-monorail/components/site-header.marko (hunk 1 above)
Comment (B77Mills) [reply packages/marko-web-theme-monorail/components/site-header.marko]: @solocommand  Indentation is fixed and i removed the block wrapper.

**Pull request merged**